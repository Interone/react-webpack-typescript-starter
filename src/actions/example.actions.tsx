import { ExampleState } from '@reducers/example.reducer';

export enum ExampleActionTypes {
    IS_EXAMPLE = 'EXAMPLE_IS_EXAMPLE',
}

export interface ExampleAction {
    readonly type: ExampleActionTypes;
    readonly payload: Partial<ExampleState>;
}

export function setExample(example: string): ExampleAction {
    return {
        type: ExampleActionTypes.IS_EXAMPLE,
        payload: { example },
    };
}
