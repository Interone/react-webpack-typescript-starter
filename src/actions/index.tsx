export {
    setExample,
    ExampleAction,
    ExampleActionTypes,
} from './example.actions';
