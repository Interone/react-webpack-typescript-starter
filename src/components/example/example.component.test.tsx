import * as React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { spy } from 'sinon';

import {
    default as ConnectedExampleComponent,
    ExampleComponent,
} from './example.component';
import { setExample } from '@actions/example.actions';

const createStore = configureStore();

describe(`ExampleComponent`, () => {
    it(`should render the example component`, () => {
        const store = createStore();

        const tree = renderer
            .create(<ExampleComponent {...store} example={'hello'} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`should not crash if we connect to the store`, () => {
        const store = createStore();
        const connectedExampleComponent = new ConnectedExampleComponent({
            store,
        });
        expect(connectedExampleComponent).toBeDefined();
    });

    it(`should have a default state`, () => {
        const store = createStore();
        const wrapper = mount(
            <ExampleComponent {...store} example={'hello'} />,
        );

        expect(wrapper.state()).toEqual({ input: '' });
    });

    it(`should set the state on change`, () => {
        const store = createStore();
        const wrapper = mount(
            <ExampleComponent {...store} example={'hello'} />,
        );
        const input = wrapper.find('.input');

        expect(wrapper.state().input).toEqual('');

        input.simulate('change', {
            target: { value: 'test' },
        });

        expect(wrapper.state().input).toEqual('test');
    });

    it(`should dispatch actions on send`, () => {
        const store = createStore();
        const dispatch = spy(store, 'dispatch');
        const wrapper = mount(
            <ExampleComponent {...store} example={'hello'} />,
        );
        const button = wrapper.find('.button');

        wrapper.state().input = 'mock';

        button.simulate('click');

        expect(dispatch.callCount).toEqual(1);
        expect(dispatch.calledWith(setExample('mock'))).toEqual(true);
    });
});
