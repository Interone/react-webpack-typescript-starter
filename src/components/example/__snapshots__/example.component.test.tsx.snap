// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent should render the example component 1`] = `
<div>
  <h1>
    hello
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="button"
      onClick={[Function]}
      type="button"
      value="click me"
    />
  </form>
</div>
`;
